storage/storage.go

```diff
@@ -62,4 +64,11 @@ type Sealer interface {
 
 	// Removes all data associated with the specified sector
 	Remove(ctx context.Context, sector SectorRef) error
+
+	// Generate snap deals replica update and return proof
+	ReplicaUpdate(ctx context.Context, sector SectorRef, pieces []abi.PieceInfo) (ReplicaUpdateProof, error)
```

Comment:
This is all I think we'll need.  Not sure about the names or need for new proof type.